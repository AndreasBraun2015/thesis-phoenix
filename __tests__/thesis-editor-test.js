import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ThesisEditor from '../web/static/js/components/thesis-editor.js'

jest.mock('react-dom')

const mockExternal = (overrides={}) => {
  let title = "Doc title"
  let description = "Description"
  let redirectURL = null
  return Object.assign({
    ospryPublicKey: null,
    pageRedirectURL: null,
    template: null,
    templates: [],
    isDynamicPage: false,
    path: "/test",
    getTitle: () => title,
    setTitle: (t) => title = t,
    getDescription: () => description,
    setDescription: (desc) => description = desc,
    getRedirectURL: () => redirectURL,
    setRedirectURL: (url) => redirectURL = url,
    save: (page, contents, callback) => callback(),
    delete: (path, callback) => callback()
  }, overrides)
}

it('renders ThesisEditor', () => {
  const tree = renderer.create(<ThesisEditor external={mockExternal({})} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders ThesisEditor with an add button', () => {
  const external = mockExternal({
    templates: ['dynamic', 'other']
  })
  const tree = renderer.create(<ThesisEditor external={external} />).toJSON()
  expect(tree).toMatchSnapshot()
})
