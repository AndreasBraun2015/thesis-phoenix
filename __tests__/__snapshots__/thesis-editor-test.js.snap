// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ThesisEditor 1`] = `
<div
  id="thesis"
>
  <div
    className=" thesis-page-tools-hidden thesis-page-tools-count-5"
    id="thesis-editor"
  >
    <div
      className="thesis-button settings"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Page Settings
      </div>
      <i
        className="fa fa-wrench fa-2x"
      />
    </div>
    <div
      className="thesis-button import-export"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Import/Export/Restore Content
      </div>
      <i
        className="fa fa-exchange fa-2x"
      />
    </div>
    <div
      className="thesis-button save"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Publish Changes
      </div>
      <i
        className="fa fa-save fa-2x"
      />
    </div>
    <div
      className="thesis-button cancel"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Discard Changes
      </div>
      <i
        className="fa fa-remove fa-2x"
      />
    </div>
    <div
      className="thesis-button edit"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Edit Page
      </div>
      <i
        className="fa fa-edit fa-2x"
      />
    </div>
  </div>
  <div
    className=" thesis-page-tools-hidden thesis-page-tools-count-5"
    id="thesis-fader"
  />
  <div
    className={null}
    id="thesis-tray"
  >
    <div
      className="attribution"
    >
      <a
        href="//github.com/infinitered/thesis-phoenix"
        target="_blank"
      >
        Thesis
      </a>
       is maintained by 
      <a
        href="//infinite.red"
        target="_blank"
      >
        Infinite Red
      </a>
    </div>
  </div>
</div>
`;

exports[`renders ThesisEditor with an add button 1`] = `
<div
  id="thesis"
>
  <div
    className=" thesis-page-tools-hidden thesis-page-tools-count-6"
    id="thesis-editor"
  >
    <div
      className="thesis-button settings"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Page Settings
      </div>
      <i
        className="fa fa-wrench fa-2x"
      />
    </div>
    <div
      className="thesis-button import-export"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Import/Export/Restore Content
      </div>
      <i
        className="fa fa-exchange fa-2x"
      />
    </div>
    <div
      className="thesis-button save"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Publish Changes
      </div>
      <i
        className="fa fa-save fa-2x"
      />
    </div>
    <div
      className="thesis-button cancel"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Discard Changes
      </div>
      <i
        className="fa fa-remove fa-2x"
      />
    </div>
    <div
      className="thesis-button add"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Add New Page
      </div>
      <i
        className="fa fa-plus fa-2x"
      />
    </div>
    <div
      className="thesis-button edit"
      onClick={[Function]}
    >
      <div
        className="tooltip"
      >
        Edit Page
      </div>
      <i
        className="fa fa-edit fa-2x"
      />
    </div>
  </div>
  <div
    className=" thesis-page-tools-hidden thesis-page-tools-count-6"
    id="thesis-fader"
  />
  <div
    className={null}
    id="thesis-tray"
  >
    <div
      className="attribution"
    >
      <a
        href="//github.com/infinitered/thesis-phoenix"
        target="_blank"
      >
        Thesis
      </a>
       is maintained by 
      <a
        href="//infinite.red"
        target="_blank"
      >
        Infinite Red
      </a>
    </div>
  </div>
</div>
`;
